### ISSUE TITLE
Post Installation Tasks not Performed

### ISSUE THREAD

**Issue by GWeck** [QubesOS/qubes-issues#5003]
**Qubes OS version**
R4.0.1

**Affected component(s) or functionality**
Installation from DVD
**Brief summary**
I  tried to install Qubes R4.0.1 from DVD on two different PCs (ASUS Z170M-E D3 with Intel Core i5-6600 and HP 840 Elitebook with Intel i5-7600, both with VT-x and VT-d) on to a SAMSUNG T5 USB-SSD with 512 GiB.

In both cases, the installation went through until the reboot, without error messages.  After booting, the normal Qubes login screen appeared, but the post installation tasks (VM creation etc.) were skipped. So the resulting system just had dom0 and no VMs and is not usable.

**To Reproduce**
Install R4.0.1 (from DVD - possibly also from USB stick) onto a USB disk. Perhaps the other systems on the local disks might cause this behavior???

**Expected behavior**
After first reboot, the necessary VMs should be created before normal login.

**Actual behavior**
incomplete installation

**Screenshots**

**Additional context**
Both systems have R3.2 installed on local disks and, via multi-boot, ubuntu 18.04 and one also Windows 7.

When booting, a short message "Failed to start loading kernel modules" is displayed. The kernel that is booted is 4.8.4. systemctl does not show error if the root partition is formatted as a standard partition, and one lvm related error if it is lvm.

**Solutions you've tried**
Is there a way to perform the necessary post-installation taks manually?

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**
Qubes installation docs, bug reports on installation, but without finding suitable ones

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**

**Comment by andrewdavidwong** [comment]
> Is there a way to perform the necessary post-installation taks manually?

Yes, but I suggest asking about this part on the `qubes-users` [mailing list](https://www.qubes-os.org/support/), since it will likely involve discussing a detailed walkthrough independent of this bug.

**Comment by andrewdavidwong** [comment]
Possibly related: #3431, #2213

**Comment by GWeck** [comment]
In the mean time, I tried - just to be sure - a newly loaded installation image and removed the internal disks of one of the PCs such that there is no other system which Grub could boot. The result is still the same.

With regard to #3431, in my case qubesd.service is started (systemctl shows it to be loaded, active and running). However, /var/log/salt/minion shows also the error
   [salt.pillar      ][CRITICAL] Specified ext_pillar interface qvm_prefs is unavailable
and the other error listed there, until
   refresh_cache     'admin.vm.List'.
Then, however, there is no refrence to QUBESD_SOCKET, but instead
   qubesd_call 'Failed to connect to qubesd service: %s', str(e))
Looks as if this call is getting the wrone parameter, or none at all.

With regard to # 2213, in my case not even the templates are created.

On this same machine, I had R4.0-rc2 installed on a USB disk and updated it to 4.0, without problems.

**Comment by marmarek** [comment]
You can start the post-installation wizard with `sudo /usr/libexec/initial-setup/initial-setup-graphical`.
As to find out why it wasn't started automatically, take a look at initial-setup related entries in `journalctl`.

**Comment by GWeck** [comment]
The post-installation wizard terminated with a message "Error in kickstart file /root/anaconda-ks.cfg at line 30 - invalid parameter value --size=927.985.583984375".
This line and the following two lines had the size value as floating point values instead of as integers.
In the setup, I had specified manual partitioning with 3 partitions (/boot as 1GB, / as 250GB, and swap as 20GB). These values apparently got translated to 927.985.583984375, 238396.3623046875 and 19071.708984375, respectively. Dropping the fraction parts from these values in the kickstart file, i.e. changing them to 927, 238396 and 19071, respectively, allowed to execute the post-installation wizard without any error.

So I suppose, this was the problem.
Perhaps it might be helpful to provie information on how to specify partition sizes to the installer.

@andrewdavidwong: I posted a short not at qubes-users of the problem and its solution. I hope this wil help.

**Comment by marmarek** [comment]
This is clearly a bug in the installer, it should not write a kickstart file it can't load anymore.

Are there really two dots in the first number? This looks very wrong.

**Comment by GWeck** [comment]
The two dots are a typing error, the number in the kickstart file was 927.98583984375.

Somehow, these numbers are a bit enigmatic. Even if I compute the number of GB *(1000/1024)^3, I dont'get them (1 GB = 931.3225746154785 * 1024*1024*1024 Byte). I suppose, this comes from the sad difference between marketing and technical GB, which often has caused trouble.

**Comment by GWeck** [comment]
The installer reports different results in sizes, depending on how the desired size is specified:
1 GiB -> 1023.98 MiB
1 GB  -> 927.99 MiB (the value used in my case)
1024 MiB -> 1023.98 MiB
1024 MB -> 959.99 MiB
I suppose that the values for GiB and MiB are the inteded ones. It might be good if GB and MB are rejected or at least cause a warning.

**Comment by GWeck** [comment]
The error reappears in the setup of R4.1, this time in the opposite direction. I specified the partition sizes in GiB, and the values stored in the kickstart file /root/anaconda-ks.cfg were in MiB, so 1 GiB became 1023.98, and so on. Truncating the values to their integer part allowed to execute he second phase of the installation.
It might be a good idea to just store these integer values into the kickstart file, as this second phase cannot change the partition sizes.

**Comment by GWeck** [comment]
The error reappeared on installing Qubes R4.1, this time in a slightly different form: I specified the partition sizes in GiB, and the values got stored in the kickstart file `/root/aconda-ks-cfg` in MiB. So 1 GiB became 1023.98, and so on, and the second phase of the installation terminated with error, as it expected an integer value.
Truncating these values in the kickstart file to their integer value allowed to perform the second phase of the installation.
It might be a good idea to just store this integer part of the value in the kickstart file, as the second part of the installation cannot change the partition sizes at all.

**Comment by GWeck** [comment]
Some comments on installing R4.1 from the 23-Dec ISO, which might help in development:

- The ISO, when copied to a USB stick via Rufus on a Windows 7 system, is not bootable. This can easily be fixed by setting its boot partition to _Active_.

- Checking the integrity of the istallation mdia before installing aborts during the check, just saying the process terminates. I am not sure whether the media was faulty, but the installation worked if no media check was requested.

- The templates coming with the installation and their associated AppVMs seem to be o.k. It was even possible to restore an AppVM based on Fedora 30 from backup (from R4.0) and get a Windows application running in it via Wine.

- Restoring templates from backups made in R4.0 is possible, but the resulting VMs (templates as well as corresponding AppVMs) only show completely black windows. Only the window frame is colored as expected. This was tested with centos-7 and ubuntu bionic beaver. The log file `/var/log/libvirt/libxl/libxl-driver.log` contains the message:
`2019-12-28 12:28:24.021+0000: libxl: libxl_console.c:108:libxl_console_get_tty: Domain 20:Unable to read console tty path '/local/domain/20/console/tty': Function not implemented`

- Starting a VM based on Windows 7 (with QWT 4.0 installed) was unsuccessful. These VMs (template as well as correspondig AppVM) could not be started at all. Dom0 showed an error message:
`Error starting Qube! Start failed: libxenlight failed to create new domain '<VMname>'`. See the log entries in `/var/log/libvirt/libxl/libxl-driver.log`:
[libxl-driver.log](https://github.com/QubesOS/qubes-issues/files/4007018/libxl-driver.log)

- A standalone HVM based on centos-8 worked without problems.

**Comment by GWeck** [comment]
I tried now the january version of R4.1. The results are similar. There are some additional observations:

- The issue with non-integer values in the kickstart files, resulting in abort of phase 2 of the installation, remains, but can be quickly solved by editing the kickstart file and starting phse 2 manually.

- A standalone HVM based on Q4OS (a debian distribution with Trinity desktop), restored from an R4.0 backup, worked without problems.

- Restoring the fedora-30 template from an R4.0 backup was possible. The template as well as an AppVM based on this template started, but any application windows in both VMs showed only a changing pattern, probably meaning that the display is not synchronized.

- Restoring the working Windows 10 template and AppVM from R4.0 (using QWT 4.0.1.3 with pre-installed Xen drivers xenbus and xenvbd version 9.0.0 - see #5462) resulted in VMs that could be started and connected to qrexec, such that Qubes showed them as running. The VMs could be used normally, but became unrespondig as soon as data transfer to/from anotther VM via clipboard or file transfer was attempted. The only recovery was restarting these VMs.

**Comment by GWeck** [comment]
With version R4.0.3-rc1, the post-installation task is still aborted if during installation manual partitioning is selected and the partition sizes are specified in GB. The cause are still invalid size values in the kickstart file `/root/anaconda-ks.cfg`. If these values are changed to nearby integer values, the second phase of the installation can be started manually, like specified above by @marmarek.

If the partition sizes are given in GiB, suitable values are stored in the kickstart file, and the second phase of the installation starts normally.

With the current installation media for R4.1, any values for the partition size lead to incorrect values in the kickstart file, no matter if the values are given in GB, GiB or MiB.

**Comment by GWeck** [comment]
The situation is the same for R4.0.3. The offending lines in the kickstart file are those beginning with `part`. The following line beginning with `logvol` are correct.

**Comment by marmarek** [comment]
I've tried to reproduce it with R4.1 but it works for me - see https://openqa.qubes-os.org/tests/5654
Can you tell exactly what partition layout you've created?

**Comment by GWeck** [comment]
I used two layouts: `/boot` with 1 GiB, `/` with 250 GiB and `swap `with 20 GiB; the other layout was `/boot` with 1 GB, `/` with 250 GB and `swap `with 20 GB.

In both cases, the entries in `anaconda-ks.cfg` contained values with long decimal parts, like 1 GiB -> 1023.98 MiB, 1 GB -> 927.99 MiB (but with about 8 decimal places). I wil repeat one of the tests and post the kickstart file here, if I manage to extract it.

**Comment by marmarek** [comment]
Besides actual numbers (1, 35, 1), I've tried the same and it works for me...
Is it with LVM thin provisioning?
Can you compare it on [this screenshot](https://openqa.qubes-os.org/tests/5654#step/install_partitioning_custom_size/13)?

**Comment by GWeck** [comment]
I retried it now with the GiB values `/boot` with 1 GiB, `/` with 250 GiB and `swap` with 20 GiB and got the same behaviour.
The resulting kickstart file after the first part of the istallation is: 
[anaconda-ks1.cfg.txt](https://github.com/QubesOS/qubes-issues/files/4113433/anaconda-ks1.cfg.txt)
This leads to aborting the second phase.
After editing this file to 
[anaconda-ks2.cfg.txt](https://github.com/QubesOS/qubes-issues/files/4113436/anaconda-ks2.cfg.txt)
the second phase of the installation can be started manually.
After comppleting the installation, the kickstart file looks the same:
[anaconda-ks3.cfg.txt](https://github.com/QubesOS/qubes-issues/files/4113438/anaconda-ks3.cfg.txt)
I used thin provisioning; the reported disk size was 465,76 GiB. Apart from the values, your screen shot looked the same, but when I entered 1 GiB for the `/boot` partition, the installer changed this to 1023.98 MiB, and the same for the `/` partition. The `swap` partition retained its integer value.

One idea: I installed using installation language German and the German keyboard layout. Could this be the cause? In German, dot and comma are interchanged.

**Comment by marmarek** [comment]
> One idea: I installed using installation language German and the German keyboard layout. Could this be the cause? In German, dot and comma are interchanged.

Yes, that could be it! Let me check.

**Comment by GWeck** [comment]
On the running system, I tried now to re-execute phase 2 of the setup (after renaming `initial-setup-ks.cfg`, so that the parameters were taken again from `anaconda-ks.cfg`).

No matter if I replaced the dot in the `part` instructions with comma or not, the values were rejected. Only if I truncated the values to their integer part, phase 2 of the installation was started.

**Comment by marmarek** [comment]
I've tried to use en_DK locale (which also has swapped dot and coma), but it still works. I had similar test for it written already. I'll try de_DE.

**Comment by GWeck** [comment]
Just to be sure, I tried it now with English (US) as installation language and a US keyboard - and got exactly the same results! The non-integer values in anaconda-ks.cfg are the same as in the test with the German installation language. Now I am really perplexed ...

Tomorrow I'll retry with the values of your screenshot.

**Comment by marmarek** [comment]
Maybe it's about actual sizes? I'll find some bigger test disk and try...

**Comment by GWeck** [comment]
I tried now with a diffrent disk - a 2 TB HDD, and it did not show the problems. The resulting `anaconda-ks.cfg` contained only integer values
[anaconda-ks-HDD.cfg.txt](https://github.com/QubesOS/qubes-issues/files/4122089/anaconda-ks-HDD.cfg.txt)
and so phase 2 of the installation started.

With the SSD that I used so far, the configuration page shows the strange behaviour that the partition sizes that I enter are changed to nearby values (1 GiB to 1023.98 MiB instead of 1024 MiB, and 250 GiB to 249.49 GiB), as soon as the small window is closed.

The swap partition behaves still more strangely: I entered 20 GiB, and it was changed to 16 GiB. After deleting this entry and re-entering it, the new size was displayed as 16.77 GiB, and a third attempt yielded 16.65 GiB; fourth even went to 13.45 GiB.

If I try the values from your screenshot, 953 MiB gets changed to 927,99 MiB, 30.18 GiB to 30.11 GiB, and 1024 MiB to 816 MiB, 

This was tested with the 4.0.3 kit, which behaves just like the 4.1-20200113 kit.

I tried to create these partitions on the SSD with Gparted to see if the drive had any problems, but there they were created normally with the intended sizes.

Could it be that the installer treis to read something from the drive and gets nonsense?

**Comment by marmarek** [comment]
I've tried with German language, and also got numbers replaced in the interface, with colon. But the ones produced in kickstart were just integers...

**Comment by GWeck** [comment]
Just to be sure, I tried it now on a different machine (R4.1 with SSD), and got the same results.

I suppose that the cause lies in the SSD disk. Language, Qubes version, computer all make no difference. Only using a HDD or SSD show different reactions. I wonder what the installer does with my SSD. How are the values created thatare stored in the kickstart file? Where do they come from? There must be some input additional to the values entered at the screen.

**Comment by marmarek** [comment]
Maybe you have an SSD with bigger sector size? If so, you may run into another issue: https://github.com/QubesOS/qubes-issues/issues/4974
Check `/sys/block/(device)/queue/hw_sector_size`

**Comment by brendanhoar** [comment]
That was my thought as well.

Most 4K sector-size SSDs can be configured to emulate 512 sectors (switch to "512e" mode), usually at a minimal performance cost (but not nearly the cost of the rare 4K sector-size HDDs doing the same 512 emulation). 

Most of the reason that the performance cost is minimal for SSDs has to do with the fact they already abstract a lot of the data-moving and R/M/W latencies away from the end-user.

**Comment by GWeck** [comment]
The reported sector size is 512 - the same that is reported if I attach the SSD to a Windows system.

SAMSUNG gives on its website information about using the Portable T5 device under Windows and MacOS, but not under Linux. On the other hand, if it works under MacOS, it should also under Linux. And it does work under Qubes as soon as the second initialization phase is executed. (I used it with R4.0 for some months in a production system without problems.)

Still the problem remains what the installer does and how it changes the given partition sizes and why it enters non-integer values into the kickstart file.

**Comment by brendanhoar** [comment]
Ah, I missed that, way up at the top, you stated that the problematic device was an external USB device.

In particular, the Samsung T5 is an mSATA drive behind a USB 3.0 <-> samsung-customized SATA bridge (customized to support Samsung proprietary AES encryption unlock at the bridge level, not TCG Opal nor ATA Password controlled). 

The devices are fairly common out there. I myself have a couple T3s and a couple T5s.

Is it possible the device, either due to the bridge or the non-standard drive firmware, reports back somewhat unexpected numbers via ATA IDENTIFY DEVICE or similar calls that confound the standard linux tooling used by anaconda?

Looks like community has seen issues with the predecessor T3 before: https://github.com/QubesOS/qubes-issues/issues/2097
https://groups.google.com/forum/#!topic/qubes-users/zVFjCSUk6KE

Probably the device is return information to the tooling that is contradictory?

B

**Comment by GWeck** [comment]
Yes, I think that is the cause.

I cannot be the disk size, as I installed R4.0.1 on a SATA SSD of the same size (Samsung SSD 850 EVO), and this worked correctly, as you can see from the corresponding kickstart file which contains only integer values:
[anaconda-ks-SATA.cfg.txt](https://github.com/QubesOS/qubes-issues/files/4128006/anaconda-ks-SATA.cfg.txt).

Still this seems to me an error in the installer, as even if it is confused by data coming from a disk, it should never write non-integer values into the kickstart file. For installation to proceed correcty, it would be enough if the installer simply truncated any values to their integer part before writing them into the kickstart file.

What is worrying me somewhat is that what happend here could happen with any other disk. The Samsung T5 is no extremely special disk, and if it happens here ... who knows.

**Comment by marmarek** [comment]
I absolutely agree this is a bug in the installer. But to properly fix it, I'd like to understand first what causes it, otherwise we may only hide (possibly not all of) its effect.

**Comment by GWeck** [comment]
You're right!!

**Comment by GWeck** [comment]
Here's a description of my test SSD, in case this might help. It is a screenshot - the utility has no text output.
![SamsungT5](https://user-images.githubusercontent.com/22047384/73380742-d868ee00-42c4-11ea-8e94-4431102586f6.jpg)
The partition layout is that of installing R4.1, after performing both phases of the installation.

**Comment by marmarek** [comment]
Looks normal...

**Comment by brendanhoar** [comment]
Yeah, I'm taking a look at the T5 info in hdparm and other tools and nothing is jumping out at me as unusual.

**Comment by brendanhoar** [comment]
Is there a way to turn on tracing of anaconda drive/partition commands/system calls?

**Comment by marmarek** [comment]
It produces a ton of logs in `/tmp/storage.log`, saved later in `/var/log/anaconda`.

**Comment by GWeck** [comment]
Here are the (non-empty) logs of my last installation. As far as I see (but I'm not sure), the T5 SSD is sdb, and sda is another R4.0 Qubes installation with a lot of VMs. nvme0nl is a separate SSD with an Ubuntu installation, which has nothing to do with the tests.
[anaconda.log](https://github.com/QubesOS/qubes-issues/files/4143835/anaconda.log)
[X.log](https://github.com/QubesOS/qubes-issues/files/4143836/X.log)
[dbus.log](https://github.com/QubesOS/qubes-issues/files/4143837/dbus.log)
[dnf.librepo.log](https://github.com/QubesOS/qubes-issues/files/4143838/dnf.librepo.log)
[hawkey.log](https://github.com/QubesOS/qubes-issues/files/4143839/hawkey.log)
[journal.log](https://github.com/QubesOS/qubes-issues/files/4143840/journal.log)
[ks-script-5jjenc9z.log](https://github.com/QubesOS/qubes-issues/files/4143841/ks-script-5jjenc9z.log)
[ks-script-d07f6pt9.log](https://github.com/QubesOS/qubes-issues/files/4143842/ks-script-d07f6pt9.log)
[ks-script-y90of9di.log](https://github.com/QubesOS/qubes-issues/files/4143843/ks-script-y90of9di.log)
[lvm.log](https://github.com/QubesOS/qubes-issues/files/4143844/lvm.log)
[packaging.log](https://github.com/QubesOS/qubes-issues/files/4143845/packaging.log)
[program.log](https://github.com/QubesOS/qubes-issues/files/4143846/program.log)
[storage.log](https://github.com/QubesOS/qubes-issues/files/4143847/storage.log)
I hope you'll find something of importance there. If you need more inforamation, please tell me, and I will try to provide it.

**Comment by GWeck** [comment]
I checked now with the current R4.1 test version (`Qubes-4.1-20200614-x86_64.iso`), and it still behaves the same way, i.e. the second phase of the installation aborts because of non-integer values in `anaconda-ks.cfg` for the partition sizes, which I specified as 1 GiB for `/boot`, 250 GiB for `/`, and 20 GiB for `swap`.

Another attempt with automatich partitioning failed due to lack of space, although there were about 300 GiB free. Here it might be helpful, if automatic partitioning would show the sizes of the intended partitions before starting the installation, perhaps allowing to change these sizes.

**Comment by GWeck** [comment]
I checked with a different drive, a 64 GiB USB 2.0 disk, and there only integer values were stored in `anaconda-ks.cfg
`. However, the second phase of the installation was not started, possibly because of a timeout, since this drive was very slow. The second phase could be started manually and completed successfully, without errors.

Checking different installations, only those on the SAMSUNG USB SSD showed the queer behaviour that non-integer values for the partition sizes were entered by phase 1 into `anaconda-ks.cfg`, causing phase 2 to abort. So it seems that there is some property of this drive to cause the non-integer values to be written, but I have no idea what it is - it is just a standard T5 SSD.

(I wouldn't repeat that new test, since the USB stick and hence installation and start of Qubes on it are incredibly sloooow!)

**Comment by GWeck** [comment]
Te problem persists in R4.1 alpha from 14-Oct-2020. Still, I am not sure whether it is important, since it occurs just on this one SSD - but there must be some reason for it.

**Comment by marmarek** [comment]
I gave up looking for the source of the issue. I've added a patch that rounds the value when saving values to the kickstart file (they may remain incorrect in GUI, though). R4.0 installation image with the patch is here: https://openqa.qubes-os.org/tests/15724/asset/iso/Qubes-4.0-20210113-x86_64.iso
R4.1 is in building right now.

**Comment by brendanhoar** [comment]
My suspicion is that the custom firmware for the usb/SATA bridge or even the custom firmware on the Samsung msata SSD is returning something unexpected. The msata is a variant of the 850 model without tcg/opal but with a Samsung-specific encryption/password scheme (and perhaps other customizations). I know the 850 works fine when directly connected. It would be interesting to repeat the experiment with the msata drive attached through a more generic bridge to determine if the issue is bridge related. Alternately with the msata drive direct attached to a Mai board msata connector.

**Comment by GWeck** [comment]
@marmarek Sometimes brute force is the best - even in programming :-)

I am downloading both the 4.0 and the 4.1 version. I've ordered an SSD of the type I currently use for my tests (SAMSUNG T5 portable 500 GB), which will probably arrive tomorrow or the day after. With that, I hope I can check the new versions; I'll report as soon as its done.

@brendanhoar  So far, I connected the external SSD to an HP Elitebook 840 G4. I don't know which controller is used there. With the new disk, I'll try to repeat the tests on a system with an ASUS Z170M-E-D3 motherboard; perhaps some differences will show up.

**Comment by brendanhoar** [comment]
> @brendanhoar So far, I connected the external SSD to an HP Elitebook 840 G4. I don't know which controller is used there. With the new disk, I'll try to repeat the tests on a system with an ASUS Z170M-E-D3 motherboard; perhaps some differences will show up.

You’ll have the same issue on the ASUS most likely (well, until Marek’s workaround addresses the issue).

What I was getting at is that external USB drives are actually two devices: a standard SATA protocol device and a bridge chip that translates protocols. The bridge chipset converts from a USB based protocol (e.g. USB bulk-storage OR UASP aka USB Attached SCSI Protocol) to SATA on the drive side. Sometimes those bridge chipsets inside the external device do something unexpected when translating the two protocols (technically three with UASP!).

In addition, the SATA drive itself (well the mSATA drive, same thing logically) is a custom firmware variant of the 850 line which we know works without issue with Qubes when directly connected to SATA controllers on the motherboard. 

I’d say the issue is most likely a firmware property (perhaps defect) in the bridge chipset between the usb connector and the SATA connector...but possibly one in the custom firmware on the drive. 

I have some 2TB Samsung T5 units, one of which I’ve cracked open in the past as they were the only > 1TB mSATA drives produced and I wants to use them as an internal laptop boot drive for Qubes. Sadly, unlike the 850 units, the drive in the T5 does not support TCG/OPAL, which I depend on.

If I can dig out the cracked open unit and one of the unsullied units to see if I can do A/B tests.

I’d be looking to see if the USB-attached T5 shows any differences with a direct attached T5 drive and perhaps if either shows significant differences with an 850 direct attached drive.

**Comment by GWeck** [comment]
I see - I thought the problem was on the PC side. But perhaps I still could find a difference, since the newly bought SSD is new, while the one I am currently using is nearly two yearts old. Maybe SAMSUNG has changed the firmware in the meantime. We'll see...

**Comment by brendanhoar** [comment]
Good point: Samsung has released several firmware updates to the T5 device, typically available for installation via a Windows based application that Samsung distributes.

**Comment by GWeck** [comment]
@brendanhoar I got a new T5 drive, but it behaves identical to the old one. I could not repeat the test on the machine with the ASUS mainboard, because there were problems with the EFI setup, but I don't think that these tests would show something new.

@marmarek The test with R4.0 was similar to the R4.1 one (see #6329), with some minor differences: The file `/root/anaconda-ks.cfg` still contained a non integer value for the size of the boot partition (the same one as for R4.1). For the root partion, I chose 240 GiB and not 250 GiB as in R4.1, and here the size was given as an integer value in `/root/anaconda-ks.cfg`! The file `/root/initial-setup-ks.cfg` contains integer values for both partition sizes. Pool creation and allocation was correct, like in R4.1.

Hopefully this points to the source of the size problem!

**Comment by andrewdavidwong** [comment]
Closing this as resolved per https://github.com/QubesOS/qubes-issues/issues/6329#issuecomment-762316601. If you believe the issue is not yet resolved, or if anyone is still affected by this issue, please leave a comment, and we'll be happy to reopen this. Thank you.
